realm/src/main/java/io/realm/RealmList.java

```diff
@@ -124,8 +195,10 @@ public E first() {
      * @return The last object
      */
     public E last() {
-        if (!view.isEmpty()) {
-            return get(size()-1);
+        if (managedMode && !view.isEmpty()) {
+            return get((int) view.size() -1);
+        } else {
+            nonManagedList.get(nonManagedList.size() - 1);
```

Comment:
if size() is 0 you get an exception here. Looks like testcases are missing for empty list.